presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoStatement.java

```diff
@@ -46,10 +48,15 @@ public ResultSet executeQuery(String sql)
             throws SQLException
     {
         try {
-            ResultSet result = new PrestoResultSet(connection().startQuery(sql));
+            StatementClient statementClient = connection().startQuery(sql);
+            ResultSet result = new PrestoResultSet(statementClient);
             currentResult.set(result);
+            checkSetOrResetSession(statementClient);
```

Comment:
Strangely when I do checkSetOrResetSession before creating the ResultSet, tests do not pass.

I double check, processResponse() which puts into setSessionProperties and resetSessionProperties invoked in StatmentClient's constructor. It should be related with ResultSet
I think I am missing something here, what do you think? @electrum 

I also add expectedExceptionsMessageRegExp to tests to be sure that exception is thrown by checkSetOrResetSession and tests passed.